src/PiggyCustomEnchants/EventListener.php

```diff
@@ -393,7 +397,7 @@ public function onDataPacketReceive(DataPacketReceiveEvent $event)
     public function checkGlobalEnchants(Player $damager, Entity $entity = null, Event $event)
     {
         //TODO: Check to make sure you can use enchant with item
-        if ($event instanceof EntityDamageEvent) {
+        if ($event instanceof EntityDamageEvent and $entity instanceof Living) {
```

Comment:
Why should the check be here? Regardless if the entity is an instanceof Living, some of these enchants should still function. (Lifesteal, Vampire)

Reply:
This is found in `checkGlobalEnchants()` which has public access and a default value set of `$entity = null`.  No checks previous to the point shown here ensure the Entity is not null and any part of this method that uses `$entity->hasEffect()` or `$entity->addEffect()` requires that the entity be an instance of `Living` as that is the earliest Entity that has those options.

Would it be preferred to move options such as Lifesteal and Vampire outside the living check as so:
``` if($entity instanceof Living) {
             //Do Effect methods (ie Blind, Poision, Crippling Strike, etc)
            }
            //Do non Effect options (ie Lifesteal, Gooey, Vampire, etc)```

Reply:
Yes, move any that don't use the entity variable.

Reply:
Fixed in:
https://github.com/DaPigGuy/PiggyCustomEnchants/pull/122/commits/f332b81334cb961deb2b42984fa1caf2e22b3c73